osler/inventory/views.py

```diff
@@ -110,12 +118,21 @@ def form_valid(self, form):
         df.save()
         return redirect('inventory:drug-list')
 
+
 def drug_dispense(request):
     pk = request.POST['pk']
     num = request.POST['num']
-    drug = get_object_or_404(models.Drug, id=pk)
+    patient_pk = request.POST['patient_pk']
+    drug = models.Drug.objects.get(pk=pk)
+    patient = Patient.objects.get(pk=patient_pk)
     can_dispense = drug.can_dispense(int(num))
+
     if can_dispense:
+        models.DispenseHistory.objects.create(drug=drug,
```

Comment:
There's a concept in django about "thin views and thick models," and so I would be tempted to have `Drug.dispsense(num, patient)` and it would do this check for you and throw an exception if the drug can't be dispensed. That makes it much easier to test, because then you can just call `drug.dispense` in the test and check for the error or not, without going through the whole rigamarole of posting to the url and so forth.